src/plugins/opencl_support/src/OpenCLSupportPlugin.cpp

```diff
@@ -292,16 +293,20 @@ void OpenCLSupportPlugin::unregisterAvailableGpus() {
 }
 
 void OpenCLSupportPlugin::loadGpusSettings() {
-    Settings * s = AppContext::getSettings();
-    foreach( OpenCLGpuModel * m, gpus ) {
-        QString key = OPENCL_GPU_REGISTRY_SETTINGS_GPU_SPECIFIC +
-            QString::number(m->getId()) + OPENCL_GPU_SETTINGS_ENABLED;
-        QVariant enabled_v = s->getValue( key );
-        if( !enabled_v.isNull() ) {
-            m->setEnabled( enabled_v.toBool() );
-        } else {
-            m->setEnabled( true );
-        }
+    Settings* s = AppContext::getSettings();
+    QString enabledGpu = s->getValue(OPENCL_GPU_REGISTRY_SETTINGS_GPU_ENABLED, QVariant()).toString();
+    CHECK_EXT(!enabledGpu.isEmpty(), gpus.first()->setEnabled(true), );
```

Comment:
"gpus" list could be empty. Do not call "first" method without check.

Reply:
Technically - yes, but literally - no, so, I think, SAFE_POINT is required.